### ISSUE TITLE
Text search on identi.ca

### ISSUE THREAD

**Issue by GoogleCodeExporter** [KGIII/pino-twitter#141]
```
Hi,

When cliking on a search link on an identi.ca notice, the brower calls an
url like 'http://identi.ca/search/notice?q= #searchterm' instead of
'http://identi.ca/search/notice?q=searchterm' (note the extra space and the
leading #)

Cheers,
Julien
```

Original issue reported on code.google.com by `Valro...@gmail.com` on 22 Mar 2010 at 8:06

**Comment by GoogleCodeExporter** [comment]
Original comment by `tro...@gmail.com` on 22 Mar 2010 at 9:23
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Hi,

I hae just understood #text format is meant to be a tag link, which means it 
should
point to http://identi.ca/tags/text

Cheers,
Julien
```

Original comment by `Valro...@gmail.com` on 9 May 2010 at 1:50

**Comment by GoogleCodeExporter** [comment]
```
What ValroffJ points is true, however the real link for #text is
http://identi.ca/tag/text (note that it's /tag/ instead of /tags/)

pino is great, i think i just started to love it :)
```

Original comment by `javierdi...@gmail.com` on 17 May 2010 at 4:35

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Here are two patches to fix this issue.

The first only fix the url to identi.ca tags, while the second fixes the regex 
for
search w/ twitter and tags w/ identi.ca

Cheers,
Julien

```

Original comment by `Valro...@gmail.com` on 23 May 2010 at 1:35

Attachments:
- [003_identica-tags.diff](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-141/comment-5/003_identica-tags.diff)
- [004_fix-search-regex.diff](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-141/comment-5/004_fix-search-regex.diff)

**Comment by GoogleCodeExporter** [comment]
```
Another issue with the regex (not only the one for tags/search, but also at 
least for
identi.ca !groups and @nicknames): when the symbol (ie. #, ! or @) is not the 
first
character of the line or doesn't follow a whitespace character (^|\\s), the text
isn't linked.

As an example, see http://identi.ca/notice/33275128 (the tag #review follows 
"(")

Cheers,
Julien
```

Original comment by `Valro...@gmail.com` on 23 May 2010 at 9:07

**Comment by GoogleCodeExporter** [comment]
```
Here is another example which is not uncommon in French: 
http://identi.ca/notice/33976553

Cheers,
Julien
```

Original comment by `Valro...@gmail.com` on 29 May 2010 at 6:27

**Comment by GoogleCodeExporter** [comment]
```
Here is a new patch which fixes this.
I think the same should be done for the @nickname regex

Cheers,
Julien
```

Original comment by `Valro...@gmail.com` on 29 May 2010 at 6:37

Attachments:
- [004_fix-search-regex.diff](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-141/comment-8/004_fix-search-regex.diff)

**Comment by GoogleCodeExporter** [comment]
```
The last patch breaks some urls containing #

I attach a new proposal which would need to be completed by some more symbols I 
guess...

Cheers,
Julien
```

Original comment by `Valro...@gmail.com` on 2 Jun 2010 at 6:48

Attachments:
- [003_fix-search-regex.diff](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-141/comment-9/003_fix-search-regex.diff)

**Comment by GoogleCodeExporter** [comment]
```
Closed by 8926723fe020 revision.
```

Original comment by `tro...@gmail.com` on 3 Jul 2010 at 11:02
- Changed state: **Fixed**
